### PR TITLE
fix: add collapse/expand to scorecard description

### PR DESCRIPTION
**Pull request by FoodProduct**
## Change description

Adds a "Show more" / "Show less" to long scorecard descriptions on the Scorecards page.

### Before
Screen Shot 2022-06-06 at 1 19 32 PM

### After
Screen Shot 2022-06-06 at 1 18 50 PM

## Type of change

- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Improvement (improves codebase without changing functionality)

## Related issues

N/A

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: src/components/Scorecards/ScorecardCard/ScorecardCard.tsx [hunk 1]
```diff
@@ -28,9 +28,10 @@ export const ScorecardCard = ({ scorecard }: ScorecardCardProps) => {
 
   return (
     <ListCard
-      name={scorecard.name}
       creatorName={scorecard.creator.name}
       description={scorecard.description}
+      name={scorecard.name}
+      truncateToCharacters={200}
```
Comment (FoodProduct) [inline src/components/Scorecards/ScorecardCard/ScorecardCard.tsx:34]: This limit is somewhat arbitrary (it's based on what we use in the Cortex web app) -- let me know if you have any suggestions!

File: src/components/ListCard/ListCard.tsx [hunk 2]
```diff
@@ -13,36 +13,67 @@
...
   ItemCardHeader,
   MarkdownContent,
 } from '@backstage/core-components';
+import { BackstageTheme } from '@backstage/theme';
+
+const useStyles = makeStyles<BackstageTheme>(_ => ({
+  linkButton: {
+    backgroundColor: 'transparent',
+    border: 'none',
+    color: '#2E77D0',
```
Comment (FoodProduct) [inline src/components/ListCard/ListCard.tsx:29]: Is there a better way to get the "primary" color (this matches `Button color="primary"`)

Comment (FoodProduct) [reply src/components/ListCard/ListCard.tsx]: Figured it out!

File: src/components/ListCard/ListCard.tsx [hunk 3]
```diff
@@ -13,36 +13,68 @@
...
       <CardMedia>
         <ItemCardHeader title={name} subtitle={`By ${creatorName}`} />
       </CardMedia>
       {description && (
         <CardContent>
-          <MarkdownContent content={description} />
+          <MarkdownContent content={descriptionToShow ?? ''} />
+          {
+            truncateToCharacters && description.length > truncateToCharacters && (
+              // <Button to={'#'} onClick={() => setIsExpanded(!isExpanded)}>{isExpanded ? 'Less' : 'More'}</Button>
```
Comment (nikhilunni) [inline src/components/ListCard/ListCard.tsx:74]: 👀

**Review by nikhilunni — DISMISSED** [review]
looks great! just one dangling comment

File: src/components/ListCard/ListCard.tsx (hunk 3 above)
Comment (FoodProduct) [reply src/components/ListCard/ListCard.tsx]: Oops 🙈. Thanks for cathing.

File: src/components/ListCard/ListCard.tsx [hunk 4]
```diff
@@ -13,36 +13,67 @@
...
   ItemCardHeader,
   MarkdownContent,
 } from '@backstage/core-components';
+import { BackstageTheme } from '@backstage/theme';
+
+const useStyles = makeStyles<BackstageTheme>(styles => ({
+  linkButton: {
+    backgroundColor: 'transparent',
+    border: 'none',
+    color: styles.palette.primary.main,
```
Comment (cristinabuenahora) [inline src/components/ListCard/ListCard.tsx:29]: 😸

File: src/components/ListCard/ListCard.tsx [hunk 5]
```diff
@@ -13,36 +13,67 @@
...
+
   return (
     <Card>
       <CardMedia>
         <ItemCardHeader title={name} subtitle={`By ${creatorName}`} />
       </CardMedia>
       {description && (
         <CardContent>
-          <MarkdownContent content={description} />
+          <MarkdownContent content={descriptionToShow ?? ''} />
```
Comment (cristinabuenahora) [inline src/components/ListCard/ListCard.tsx:71]: I could see wanting a component ExpandableText or something like this to use in other places as well but up to you.

File: src/components/ListCard/ListCard.tsx [hunk 6]
```diff
@@ -13,36 +13,67 @@
...
+      return description.substring(0, truncateToCharacters) + '...';
+    }
+
+    return description;
+  }, [description, isExpanded, truncateToCharacters]);
+
   return (
     <Card>
       <CardMedia>
         <ItemCardHeader title={name} subtitle={`By ${creatorName}`} />
```
Comment (cristinabuenahora) [inline src/components/ListCard/ListCard.tsx:67]: @nikhilunni should we remove the creator? we did in Cortex

**Review by cristinabuenahora — DISMISSED** [review]
LGTM but maybe wait for @nikhilunni to review too

File: src/components/ListCard/ListCard.tsx (hunk 6 above)
Comment (nikhilunni) [reply src/components/ListCard/ListCard.tsx:67]: oh yeah we should, totally forgot about that

Comment (FoodProduct) [reply src/components/ListCard/ListCard.tsx]: Of course -- updated 👍

File: src/components/ListCard/ListCard.tsx (hunk 5 above)
Comment (FoodProduct) [reply src/components/ListCard/ListCard.tsx:69]: I think I'll defer that to the future for now -- good idea though!

**Pull request merged**